internal/rsyncd/rsyncd.go

```diff
@@ -25,35 +25,32 @@ type sendTransfer struct {
 }
 
 func NewServer(modules ...config.Module) *Server {
-	moduleMap := map[string]config.Module{}
-	for _, module := range modules {
-		moduleMap[module.Name] = module
-	}
-
-	return &Server{moduleMap}
+	return &Server{modules}
 }
 
 type Server struct {
-	modules map[string]config.Module
+	modules []config.Module
 }
 
 func (s *Server) getModule(requestedModule string) (config.Module, error) {
-	m, ok := s.modules[requestedModule]
-	if !ok {
-		return config.Module{}, fmt.Errorf("no such module")
+	for _, mod := range s.modules {
+		if mod.Name == requestedModule {
+			return mod, nil
+		}
 	}
-	return m, nil
+
+	return config.Module{}, fmt.Errorf("no such module: %s", requestedModule)
```

Comment:
Heads up: I added name of the module that is not found, I figured it was missing since it was using `fmt.Errorf()`, i.e. format string with which there is an expectation that one should put something *dynamic* in there.

If we really want to return a static error message, then `errors.New()` is more semantic, WDYT?

Let me know if this is not good. OTOH, the only call-site doesn't consume the error message (it just checks for non-nil error)

Reply:
Yeah, we can add the name of the module here. Maybe it’ll surface in unit tests, in which case it’d be good to have it :)